spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/event/AfterDeleteEvent.java

```diff
@@ -0,0 +1,45 @@
+/*
```

Comment:
Just to clarify, what is the need for the 4 classes `BeforeSaveEvent`, `AfterSaveEvent`, etc. over just instantiating the parent class?

Reply:
I suppose you could have a setting in `SaveEvent` to indicate if it was a "before" or an "after" event. However it looks like the Spring Data way is to use separate classes for the before and after. In future cases they might actually hold different data, too.